tensorflow_addons/rnn/cell.py

```diff
@@ -366,3 +366,471 @@ def _create_norm_layer(self, name):
             gamma_initializer=self.norm_gamma_initializer,
             epsilon=self.norm_epsilon,
             name=name)
+
+
+@tf.keras.utils.register_keras_serializable(package='Addons')
+# class LayernormSimpleRNNCell(SimpleRNNCell, LayerNormalization):
+class LayernormSimpleRNNCell(keras.layers.SimpleRNNCell):
+    """Cell class for LayernormSimpleRNN.
+
+    Motivation:
+    - Drop-In Replacement for keras.layers.SimpleRNNCell
+    - demonstrate how to add keras.layers.LayerNormalization
+       to all RNNs by introducing the `use_layernorm` argument
+
+    References:
+    [1] Ba, Jimmy Lei, Jamie Ryan Kiros, and Geoffrey E. Hinton.
+        "Layer Normalization." ArXiv:1607.06450 [Cs, Stat],
+        July 21, 2016. http://arxiv.org/abs/1607.06450
+
+    Arguments:
+      units: Positive integer, dimensionality of the output space.
+      activation: Activation function to use.
+        Default: hyperbolic tangent (`tanh`).
+        If you pass `None`, no activation is applied
+        (ie. "linear" activation: `a(x) = x`).
+      use_bias: Boolean, (default `True`), whether the layer uses a bias
+        vector.
+      use_layernorm: Boolean, (default `True`), whether to apply layer
+        normalization (scaling only).
+      layernorm_epsilon: Float, (default `1e-5`), Small float added to variance
+        to avoid dividing by zero.
+      kernel_initializer: Initializer for the `kernel` weights matrix,
+        used for the linear transformation of the inputs. Default:
+        `glorot_uniform`.
+      recurrent_initializer: Initializer for the `recurrent_kernel`
+        weights matrix, used for the linear transformation of the recurrent
+        state. Default: `orthogonal`.
+      bias_initializer: Initializer for the bias vector (`use_bias=True`) or
+         for the beta vector in layer normalization (`use_layernorm=True`).
+         Default: `zeros`.
+      gamma_initializer: Initializer for the gamma vector of the layer
+         normalization layer (`use_layernorm=True`). Default: `ones`.
+      kernel_regularizer: Regularizer function applied to the `kernel` weights
+        matrix. Default: `None`.
+      recurrent_regularizer: Regularizer function applied to the
+        `recurrent_kernel` weights matrix. Default: `None`.
+      bias_regularizer: Regularizer function applied to the bias vector
+         (`use_bias=True`) or for the beta vector of the layer normalization
+         layer (`use_layernorm=True`). Default: `None`.
+      gamma_regularizer: Regularizer function applied to the gamma vector
```

Comment:
Not sure where the gamma regularizer and constrant are useful or not. I chose to not expose those two in the layernorm lstm, so that we don't have a big number of params that confuse users.

Reply:
It's for training of the scaling parameter in `LayerNormalization`. 

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/layers/normalization.py
line 1033 - 1040

Why to set `self.gamma_constraint`?

- When your scaling parameters shrinks towards zero (e.g., other weights increase, too unscaled inputs), or becomes too large
- You know from experience with the problem, that the scaling parameter always was in a certain range

Why to set `self.gamma_regularizer`?

- When your scaling parameter explodes (e.g., input values and hidden state values are too small to generate new hidden values that are big enough)

The gamma constraint and regularizer are hyperparameters a user could try when model training terminates with weird scaling parameters.